### DISCUSSION TITLE
Adding an (optional) semantic layer to Typst

### DISCUSSION THREAD

**Discussion by markdoerr** [typst/typst#4479] [Ideas]
## Overview and Motivation

There is currently a large movement in **science** to make scientific texts machine "understandable" / interpretable and actionable (e.g. in the *FAIR data movement*). 
A common path many scientists and developers follow is to relate each well defined *term* in a (scientific) text to a corresponding term from an *Ontology* or *Taxonomy*.  By doing so, the text is enriched by its context and each word receives a well defined meaning (=semantics).
This enhanced text representation would enable computer programs to "understand" the context of a text and concrete meaning of a certain word in a sentence. This semantic information could, e.g., be used to display further information about a certain concept to the user, or to enable machine learning or AI to interpret the content of a text much more accurately. 

Adding an (optional) semantic layer to **Typst** would make **Typst** the ultimate text format for modern scientific texts.

## Proposal
Since it is not *ab-initio* clear, how the best implementation for this challanging feature looks like, I propose to create  
a new repository in the *Typst github group*, e.g. called "typst-semantics" to be used for exploration and discussion.
Within this repository, different implementations could be discussed and compared.
Branches could be used to illustrate a certain implementation as a Proof-of-Concept.

An example repo - to provide some inspirations - was created at: [https://github.com/SciTXT/typst-semantics](https://github.com/SciTXT/typst-semantics).

## Illustrating Example of what could be linked

The following biomedical *Typst* text shall illustrate **what could be linked** not **how the implementation should look like**. It contains sample references (in this case as CURIs) to external resources that define or explain the terms in more details
(the given Links in brackets should only be shown to the user, when she/he moves over a certain term, they should ideally not appear while editing a **Typst** document - or they should be at least hide-able).

```
= Oxytocin

*Oxytocin* is a [peptide hormone](wikidata:Peptide_hormone) and [neuropeptide](wikidata:Neuropeptide) 
normally produced in the [hypothalamus](wikidata:Hypothalamus) and released by the [posterior pituitary](ncit:Posterior_pituitary).[...] 

*Oxytocin* is released into the bloodstream as a hormone in response to [sexual activity](https://en.wikipedia.org/wiki/Human_sexual_activity) and during childbirth [...]
```

**Comment by jamesrswift** [comment]
Is this not just references with extra steps?

**Reply by markdoerr** [reply]
No @JamesxX, 
the example above should only illustrate, **what** information could be linked, just to make the benefit of semantic texts more obvious.
The implementation (so the **how**) is very open to discussion.
